autoPlay.user.js

```diff
@@ -682,20 +694,32 @@ function useMetalDetectorIfRelevant() {
 			return;
 		}
 
-		var enemy = g_Minigame.m_CurrentScene.GetEnemy(g_Minigame.m_CurrentScene.m_rgPlayerData.current_lane, g_Minigame.m_CurrentScene.m_rgPlayerData.target);
-		// check if current target is a boss, otherwise we won't use metal detector
-		if (enemy && enemy.m_data.type == ENEMY_TYPE.BOSS) {
-			var enemyBossHealthPercent = enemy.m_flDisplayedHP / enemy.m_data.max_hp;
-
-			if (enemyBossHealthPercent >= 0.9) { // We want sufficient time for the metal detector to be applicable
-				// Metal Detector is purchased, cooled down, and needed. Trigger it.
-				advLog('Metal Detector is purchased and cooled down, Triggering it on boss', 2);
-				triggerAbility(ABILITIES.METAL_DETECTOR);
-			}
+		if (shouldUseItemOnCurrentEnemy(ENEMY_TYPE.BOSS, 0.9)) { // We want sufficient time for the metal detector to be applicable
+			advLog('Metal Detector is purchased and cooled down, Triggering it on boss', 2);
+			triggerAbility(ABILITIES.METAL_DETECTOR);
 		}
 	}
 }
 
+function useTreasureIfRelevant() {
```

Comment:
We should handle the treasure and the metal detector the same way internally, instead of having two functions.